src/protocol/get_pfm_id.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+//! `GetPFMId` request and response.
+//!
+//! This module provides a Cerberus command that allows the querying of
+//! platform firmware manifest ID.
+
+use core::alloc::Layout;
+
+use crate::io::ReadInt as _;
+use crate::protocol::CommandType;
+
+#[cfg(feature = "arbitrary-derive")]
+use libfuzzer_sys::arbitrary::{self, Arbitrary};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+protocol_struct! {
+    /// A command for requesting the PFM Version or Platform ID.
+    ///
+    /// The encoding of the response structure diverges from that of the
+    /// specification. The spec requires that we know the original request in
+    /// order to parse the response. As of now, we will be diverging from that.
+    /// Instead, we encode and decode this with an extra byte that indicates
+    /// whether the response contains an `IdentifierType::Version` or
+    /// `IdentifierType::Platform` and the encoding of this follows that of the
+    /// request.
+    ///
+    /// The encoding is as follows:
+    ///
+    /// 1 - PFMValid (0 or 1)
+    /// 2 - IdentifierType (as encoded in the request)
+    /// 3:N - Identifier
+    ///       - If `Version` this will be exactly 4 bytes
+    ///       - If `Platform` this will be an ASCII string (not null terminated).
+    type GetPFMId;
+    const TYPE: CommandType = GetPFMId;
+
+    struct Request {
+        /// The port of the device whose PFM Id is being looked up.
+        pub port_id: u8,
+        /// The PFM Region (Active or Pending).
+        pub pfm_region: PFMRegion,
```

Comment:
`PFMRegion` -> `PfmRegion`. Acronyms in Rust are in `Rpc` case, not `RPC` case.

(Applies throughout, ofc.)